### DISCUSSION TITLE
Track train

### DISCUSSION THREAD

**Discussion by alexandrebrosse** [Alexis06030631/SNCF.js#9] [Q&A] [answered]
Hi,

Is it possible to track a train with your library ? I've got the train number + train date as input. I'd like to track the train timings and lates.

Thanks

**Comment by Alexis06030631** [comment]
Hi @alexandrebrosse !

In the current version it is not possible to do this.
**HOWEVER!** I'm currently working on your request and an update of this module will bring the functionality you're asking for!

If all goes well, I'll push the update **within the hour**!

**Reply by alexandrebrosse** [reply]
wow that's fast ! Thanks for the feedback. I assume you use the Navitia API for your lib? 

For information I tried to use  Navitia's /vehicle_journeys to track trains but it seems that a lot of disruptions are not managed by their API. 

For exemple there are no disruptions in /vehicle_journeys for today's train 6614 (currently announced with a 4h late on SNCF website)

If you can do better than I did I'll be really impressed !!

**Reply by Alexis06030631** [reply]
Indeed, I use the Navitia API, and I've just seen what's happening with your train. It's strange, but it seems that other sites have the same problem and don't display delays :( It works normally, so I have a delay displayed on my train! Keep an eye on it!

As soon as I finish the update I'll push everything and complete the doc!

**Reply by Alexis06030631** [reply, marked as answer]
### As promised! 

### The new function that lets you search for a train using its ID (headsign)

### requirements:
Update the module: `npm i sncf.js@latest`

How to use: 
```javascript
sncf.searchManager.searchTrain('HEADSING').then(data=>{
  console.log(data)
})
```

[Documentation of the new function](https://alexis06030631.github.io/SNCF.js/api/sncf.js.searchmanager.html)
